### PR TITLE
bugFix/change reward address

### PR DESCRIPTION
**Pull request by sasurobert**
Change reward address fix - auction smart contract was calling in the wrong order the staking smart contract.

### CODE REVIEW

File: integrationTests/testProcessorNodeWithMultisigner.go [hunk 1]
```diff
@@ -112,6 +113,169 @@ func CreateNodesWithNodesCoordinatorWithCacher(
...
+			)
+		}
+
+		nodesMap[shardId] = append(nodesMap[shardId], nodesList...)
+	}
+
+	return nodesMap
+}
+
+func CreateNodeWithBLSAndTxKeys(
```
Comment (iulianpascalau) [inline integrationTests/testProcessorNodeWithMultisigner.go:183]: Please add comment

File: integrationTests/testProcessorNodeWithMultisigner.go [hunk 2]
```diff
@@ -112,6 +113,169 @@ func CreateNodesWithNodesCoordinatorWithCacher(
 
 }
 
+// CreateNodesWithNodesCoordinatorAndTxKeys
```
Comment (iulianpascalau) [inline integrationTests/testProcessorNodeWithMultisigner.go:116]: please add a " -"

File: vm/systemSmartContracts/auction.go [hunk 3]
```diff
@@ -183,13 +183,15 @@ func (s *stakingAuctionSC) changeRewardAddress(args *vmcommon.ContractCallInput)
 		return vmcommon.UserError
 	}
 
+	txData := "changeRewardAddress@" + hex.EncodeToString(registrationData.RewardAddress)
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go:186]: for this, I have used something like this in genesis package
```GO
arguments := make([]string, 0, len(registrationData.BlsPubKeys) + 2)
arguments = append(arguments, changeRewardAddressFunction)
arguments = append(arguments, hex.EncodeToString(registrationData.RewardAddress))
for _, blsKey := range registrationData.BlsPubKeys {
    arguments = append(arguments, hex.EncodeToString(blsKey))
}
txData := strings.Join(arguments, "@")
```
As can be seen, "changeRewardAddress" might be a const for this.

File: vm/systemSmartContracts/auction.go [hunk 4]
```diff
@@ -183,13 +183,15 @@ func (s *stakingAuctionSC) changeRewardAddress(args *vmcommon.ContractCallInput)
...
-		if isError {
-			log.LogIfError(err)
-			return vmcommon.UserError
-		}
+		txData += "@" + hex.EncodeToString(blsKey)
+	}
+	vmOutput, err := s.executeOnStakingSC([]byte(txData))
+	isError := err != nil || vmOutput.ReturnCode != vmcommon.Ok
+	if isError {
+		log.LogIfError(err)
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go:193]: maybe log at debug level the original error from vmOutput

File: vm/systemSmartContracts/staking.go [hunk 5]
```diff
@@ -276,8 +276,7 @@ func (r *stakingSC) changeRewardAddress(args *vmcommon.ContractCallInput) vmcomm
 		return vmcommon.UserError
 	}
 
-	for i := 1; i < len(args.Arguments); i++ {
-		blsKey := args.Arguments[i]
+	for _, blsKey := range args.Arguments[1:] {
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/staking.go:279]: :1st_place_medal:

File: integrationTests/testProcessorNodeWithMultisigner.go (hunk 1 above)
Comment (sasurobert) [reply integrationTests/testProcessorNodeWithMultisigner.go:183]: done

File: integrationTests/testProcessorNodeWithMultisigner.go (hunk 2 above)
Comment (sasurobert) [reply integrationTests/testProcessorNodeWithMultisigner.go:116]: done

File: vm/systemSmartContracts/auction.go (hunk 4 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:193]: vmOutput cannot is only userError.

File: vm/systemSmartContracts/auction.go (hunk 3 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:186]: i would leave it like this. it is the same.

Comment (iulianpascalau) [reply vm/systemSmartContracts/auction.go:186]: I know, just it was clearer what you add to the arguments list.

File: vm/systemSmartContracts/auction.go (hunk 4 above)
Comment (iulianpascalau) [reply vm/systemSmartContracts/auction.go:193]: yeah, could have used vmOutput.ReturnMessage but I'm guessing the system SC does not write that.

**Pull request merged**